runtime/dancebox/src/lib.rs

```diff
@@ -784,21 +798,35 @@ impl ProvideBlockProductionCost<Runtime> for BlockProductionCost<Runtime> {
     }
 }
 
+pub struct CollatorAssignmentCost<Runtime>(PhantomData<Runtime>);
+impl ProvideCollatorAssignmentCost<Runtime> for CollatorAssignmentCost<Runtime> {
+    fn collator_assignment_cost(_para_id: &ParaId) -> (u128, Weight) {
+        (FIXED_COLLATOR_ASSIGNMENT_COST, Weight::zero())
+    }
+}
+
 parameter_types! {
     // 60 days worth of blocks
-    pub const MaxCreditsStored: BlockNumber = 60 * DAYS;
+    pub const MaxBlockProductionCreditsStored: BlockNumber = 60 * DAYS;
+    // 60 days worth of blocks
+    pub const MaxCollatorAssignmentCreditsStored: u32 = MaxBlockProductionCreditsStored::get()/Period::get();
 }
 
 impl pallet_services_payment::Config for Runtime {
     type RuntimeEvent = RuntimeEvent;
     /// Handler for fees
     type OnChargeForBlock = ();
+    type OnChargeForCollatorAssignment = ();
```

Comment:
Have you already planned for a future implementation of this handler? Is it going to be something similar to what we want to do with unspent tokens in `para_deregistered` of `ServicesPayment`? I mean sending the unspent tokens to a user-configured account (I might be missing some context and this could be a totally different situation by the way, so correct me if I'm wrong).

Reply:
no, I think the idea would be:

- unspent tokens go to a container-chain manager dictated refund address.
- collator-assignment spent tokens might involve giving some part to collators, and some other to get burnt
- block production spent tokens might only involve burning

At least the hooks let us implement this whatever way we want in the future